extension/storage/filestorage/config.go

```diff
@@ -59,18 +63,16 @@ func (cfg *Config) Validate() error {
 		dirs = []string{cfg.Directory}
 	}
 	for _, dir := range dirs {
-		info, err := os.Stat(dir)
-		if err != nil {
-			if os.IsNotExist(err) {
+		if info, err := os.Stat(dir); err != nil {
+			if !cfg.CreateDirectory && os.IsNotExist(err) {
 				return fmt.Errorf("directory must exist: %w", err)
```

Comment:
Perhaps worth updating this error message to point to the `create_directory` option?

Reply:
Done.